src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,52 @@
+import { ChangeEvent, useState } from 'react';
+import { SearchLink } from './SearchLink';
+import { getSearchWith } from '../utils/searchHelper';
+import { useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
+
+const ageParams = ['16', '17', '18', '19', '20'];
+
 export const PeopleFilters = () => {
+  const [query, setQuery] = useState('');
+  const [searchParams, setSearchParams] = useSearchParams();
+  const ages = searchParams.getAll('centuries');
+  const sex = searchParams.get('sex') || null;
+
+  const handleQueryChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const serch = getSearchWith(searchParams, {
+      query: e.target.value || null,
+    });
+
+    setSearchParams(serch);
+    setQuery(e.target.value);
+  };
+
+  const addAge = (age: string) => {
+    return ages.includes(age)
+      ? ages.filter(currAge => currAge !== age)
+      : [...ages, age];
+  };
```

Comment:
In the `addAge` function, ensure that when updating the `centuries` parameter, other existing search parameters like `sex` or `query` are preserved. This will prevent them from being unintentionally removed from the URL.